test/polygon/hop/hop.test.ts

```diff
@@ -0,0 +1,240 @@
+      expect(!!dsaWallet0.address).to.be.true;
+    });
+
+    it("Deposit MATIC & DAI into DSA wallet", async function () {
+      await wallet0.sendTransaction({
+        to: dsaWallet0.address,
+        value: ethers.utils.parseEther("10")
+      });
+      expect(await ethers.provider.getBalance(dsaWallet0.address)).to.be.gte(ethers.utils.parseEther("10"));
+
+      await addLiquidity("dai", dsaWallet0.address, ethers.utils.parseEther("10000"));
+    });
+  });
+
+  describe("Main", function () {
+    it("should migrate from L2 to L1", async function () {
+      const amount = ethers.utils.parseEther("10");
+      const bonderFee = new BigNumber(100 * 1e9)
+        .multipliedBy(150000)
+        .multipliedBy(1.5)
+        .plus(new BigNumber(amount.toString()).multipliedBy(0.18));
+      const deadline = Date.now() + 604800;
+      const getId = "0";
+
+      await token.connect(wallet0).approve(l2AmmWrapper, amount.toString());
+
+      const hDaiIdx = await saddleSwapInstance.connect(wallet0).getTokenIndex(hDai);
+      const daiIdx = await saddleSwapInstance.connect(wallet0).getTokenIndex(DAI_ADDR);
+
+      let amountOutMin = await saddleSwapInstance.connect(wallet0).calculateSwap(daiIdx, hDaiIdx, amount.toString());
+      amountOutMin = new BigNumber(amountOutMin.toString()).multipliedBy(0.99);
+
+      const params: any = [
+        DAI_ADDR,
+        l2AmmWrapper,
+        wallet0.address,
+        1,
+        amount.toString(),
+        bonderFee.toString(),
+        amountOutMin.toFixed(0),
+        deadline,
+        "0",
+        "0"
+      ];
+
+      const spells = [
+        {
+          connector: connectorName,
+          method: "bridge",
+          args: [params, getId]
+        }
+      ];
+
+      const tx = await dsaWallet0.connect(wallet0).cast(...encodeSpells(spells), wallet1.address);
+      let receipt = await tx.wait();
+      latestBlock = receipt.blockNumber;
+    });
+
+    it("Should fetch transferId from TransferSent Event", async function () {
+      const filter = l2BridgeInstance.connect(wallet0).filters.TransferSent();
+      const events = await l2BridgeInstance.connect(wallet0).queryFilter(filter, 27054896, latestBlock);
+
+      const transferSentEvent: any = events[0].args;
+
+      expect(transferSentEvent[2].toLowerCase()).to.be.equals(wallet0.address.toLowerCase());
+      console.log("TransferId", transferSentEvent[0]);
+    });
+
+    it("should migrate from L2 to L2", async function () {
+      const amount = ethers.utils.parseEther("10");
+
+      const hDaiIdx = await saddleSwapInstance.connect(wallet0).getTokenIndex(hDai);
+      const daiIdx = await saddleSwapInstance.connect(wallet0).getTokenIndex(DAI_ADDR);
+
+      let destinationAmountOutMin = await saddleSwapInstance
+        .connect(wallet0)
+        .calculateSwap(hDaiIdx, daiIdx, amount.toString());
+      destinationAmountOutMin = new BigNumber(destinationAmountOutMin.toString()).multipliedBy(0.99);
```

Comment:
@shanefontaine This too?

Reply:
👍  LGTM

Reply:
Okay, @shanefontaine, So for calculating amountOutMin and destinationAmountMin we will be using the source chain saddleSwap contract.